### ISSUE TITLE
Once user GitHub authenticated render the avatar in the activity bar

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#136889]
Once a user logins to GitHub from VS Code we could render a nice GitHub avatar instead of the soulless authentication icon in the activity bar. I personally like how Chrome and GitHub do this.

Reasons why we should do it:

* Gives confirmation to the user that she successfully logged in. Without this the experience feels half baked.
* Makes VS Code feel more personalised
* It is possible to hide it via context menu for users that dislike seeing their avatar

![Screenshot 2021-11-10 at 19 02 56](https://user-images.githubusercontent.com/1926584/141168141-86bc090b-b7b0-4d6c-a5da-27812e501141.png)

![Screenshot 2021-11-10 at 19 04 39](https://user-images.githubusercontent.com/1926584/141168447-6ca00b79-7f97-424f-9567-c3e471b1a994.png)

**Comment by TylerLeonhardt** [comment]
What would happen if they are logged in to GitHub and Microsoft accounts?

**Comment by misolori** [comment]
We did previously discussed this, I personally like it, but people on the team didn't want to look at their face all the time 🤷‍♂️ we could always add a setting to turn this off. We could also default to one of the login providers and allow a setting to set it as well.

**Comment by isidorn** [comment]
If there are multiple accounts I would say the first one or last one wins (we do the same in API in some places).
Not looking at your face the whole time -> it just has to be done with style like Chrome and GitHub. Nobody complains about those.
I think the biggest challenge is that Activity Bar icons are large and this should not be so large to not be distracting.

**Comment by misolori** [comment]
Oh I agree with you, don't get me wrong, this is just the feedback we saw in the past. FWIW, this is what it would look like:

<img width="643" alt="CleanShot 2021-11-10 at 12 48 02@2x" src="https://user-images.githubusercontent.com/35271042/141191018-57f50f8a-4209-46ea-9385-ad7ed9b490b6.png">

**Comment by isidorn** [comment]
I actually think it is pretty neat. Let me ping people on the UX channel so we hear potential negative feedback.

**Comment by isidorn** [comment]
Reasons why we should do it:
* Gives confirmation to the user that she successfully logged in. Without this the experience feels half baked.
* Makes VS Code feel more personalised
* It is possible to hide it via context menu for users that dislike seeing their avatar

**Comment by sana-ajani** [comment]
+1. Also, could give more incentive to users for logging in and syncing their settings because it gives more of a complete "profile" feel too. If I see my friend's VS Code with their picture icon, I'd be likely to ask why and do it myself to get the personal feel.

**Comment by TylerLeonhardt** [comment]
I would imagine we would need an API for this - something on an Auth Provider that is called:

```ts
export interface AuthenticationProvider {
    getAvatar(): Thenable<Uri>;
}
```

but I think we gotta be careful with that because we don't want an auth provider to simply return the url of the avatar on the internet thus having the account menu depend on the internet... ideally they download the image locally and serve it from there.

Maybe the API could return the image in a raw form but I don't think our API has done this before...

I like the idea from a personal feel... but I don't like it if they're logged in to multiple accounts.

cc @eamodio

**Comment by rebornix** [comment]
If users have two accounts, I would prefer to show both of them, one on top of another with some offset, we can always show the last one at the very top. And when an extension requests token for one account, show the account avatar at the very top. Otherwise it might look weird that I'm seeing my github avatar with `1` notification, but after clicking on it it says extensions asking for microsoft token.

With that being said, my 2 cents on if we should show it is Browsers/Websites account systems are different from ours. In browser, or github.com, you authenticate with one single primary account, and there is only one active account per instance. However in VS Code, there is no such primary account concept. I can use ms for setting sync and live share but github account for all github features. It feels weird that we promote one account over the others.

**Comment by lramos15** [comment]
I understand why people may want this, but I would definitely be someone who turns this off. To me it works on webpages because of there being different types of UI elements on the same axis as the user profile. In the activity bar it's all icons that are theme-reactive. This means my face would look awfully out of place in the UI. I also agree that multiple accounts pose a weird challenge that most sites don't have.

**Comment by misolori** [comment]
The main point of the avatar is to indicate that "you are signed in", it builds the confidence that it's working. When it's not working, your avatar would be gone and you'd see the badge that says you are signed out. Those indicators are clear.

In terms of which profile to show vs showing both, because there are many different preferences on this topic this should be user configurable. As someone who has the same profile picture everywhere, I wouldn't want to see my face twice.

**Comment by isidorn** [comment]
Great discussion, thanks for feedback.
Multi account: I agree with what has already been said. Here's a proposed solution:
* We do image diff and if we see that there are multiple different images we add a new entry "Use as Avatar" above the "Sign Out" in the submenu of each provider.
* Use the first that signed in, until the user explicitly chooses the one she wants

![Screenshot 2021-11-12 at 12 16 05](https://user-images.githubusercontent.com/1926584/141458647-c11d4a17-25eb-4228-98a2-e5cafc7e3437.png)

As for API, I agree with @TylerLeonhardt that this should not depend on the internet ideally. I do not know if returning the raw image would be in the spirit of our API.

@misolori your avatar is actually good quality and has shades of dark that go well with the dark theme. Can you do some mockups of the [default GitHub avatar](https://github.com/jvesouza) and some others from the team that are not so professional so we also get a feeling how the worse case scenario would look. Thanks!

**Comment by TylerLeonhardt** [comment]
> some others from the team that are not so professional

...

> Miguel chooses mine

![insulted](https://media.giphy.com/media/7gqN2jtvF2OIM/giphy.gif)

😛

**Comment by eamodio** [comment]
While I like the idea of having avatars (even with an option of showing it on the bar) -- I'd be a little concerned because in these other sites/apps, Chrome, Slack, GitHub, you only log in with a single user account, but in VS Code the account system isn't [just] for logging VS Code into an account, it's an account management tool/UI for both VS Code and extensions to leverage (and consolidate).

Maybe it can be split up, so that settings sync (and whatever else gets associated with a "VS Code" account, e.g. profiles?) gets elevated to a top-level login -- that would give you the single login with the avatar etc. And then move the additional accounts into a new place.

Personally, I would love to see the accounts UI move away from a menu and instead have a tailored custom fly-out to see and manage accounts (with avatars) and their usages more easily, e.g. an account overview UI. With that you could still have a combined icon (with a single "login" for vs code) and then have distinct sections for the vs code login vs other accounts.

**Comment by isidorn** [comment]
@misolori ok looks cool for all avatars. We bring this up in the UX meeting?

@eamodio makes sense. The fly-out menu would be cool, until we have that I think my [proposal from above](https://github.com/microsoft/vscode/issues/136889#issuecomment-967027000) might work just fine.

**Comment by isidorn** [comment]
After discussion in UX:
* Should we use a different codicon when a user is logged in
* We need to make showing pictures configurable, since hiding the activity bar item is too nuclear
* There might be more corner cases (like not having internet connection, some authentication not being valid) - what should we do then
* Showing an icon would give confirmation to the user that they are logged in

Based on the above I propose the following:
* We "hack" this up in Insiders so we can really try it out for a week and see if we hate it internally as a team
* If people actually turn out to like it then:
   * Let's design the API that makes sense
   * Let's add a configuration that would make this configurable
* If people do not like it:
  * Let's try to design a codicon that would show logged in state 

@TylerLeonhardt what do you think? Did I miss something?

**Comment by TylerLeonhardt** [comment]
Yes I think this is a fine plan for when we want to explore this. Maybe we can try in December.

**Comment by isidorn** [comment]
Sounds great. Let me also assign myself and let me know the best way I can help. Though let's talk in December when we find time...

**Comment by TylerLeonhardt** [comment]
What do you all think of @misolori's mockup here:
https://github.com/microsoft/vscode/issues/97168#issuecomment-628794836

**Comment by eamodio** [comment]
Oooh. I really like it except for having to hit the x to close it.

**Comment by isidorn** [comment]
@TylerLeonhardt looks great! I agree with @eamodio The whole Account and the X part can be dropped. I think it is pretty obvious without it and would make it leaner.
We still plan to render the Avatar in the activity bar but the mock just does not cover it, right?

**Comment by gjsjohnmurray** [comment]
When working in this area please cater for authentication providers with which multiple logins may exist concurrently, e.g.

![image](https://user-images.githubusercontent.com/6726799/152521303-5d572528-bbbd-4f65-9e46-78fac63f38bd.png)

**Comment by TylerLeonhardt** [comment]
@gjsjohnmurray I'm curious how this is implemented. Is it a single AuthProvider that supports multiple accounts?

**Comment by gjsjohnmurray** [comment]
@TylerLeonhardt yes, a single provider. Currently on the prerelease branch [here](https://github.com/intersystems-community/intersystems-servermanager/blob/prerelease/src/authenticationProvider.ts).

**Comment by TylerLeonhardt** [comment]
@gjsjohnmurray that's very cool. So each server has potentially different credentials so you need to log in to each individual one? How do you use this AuthProvider? How do you differentiate between sessions?

**Comment by illiaNP** [comment]
Is it possible to see it instead of the Accounts icon?
For example, let's see our Github avatar instead of the Accounts icon in our VS Code.

**Comment by RandomMonkey99** [comment]
you could try with custom css

**Comment by ShadowNineX** [comment]
Any updates?

**Comment by zifeng-yw** [comment]
<img width="845" height="536" alt="Image" src="https://github.com/user-attachments/assets/0dddf20c-8845-46f7-bd18-739ea3c5b8d0" />

From what I see in the release notes, VS Code does have this feature.
